src/background/host.ts

```diff
@@ -34,23 +36,147 @@ export class Host {
         });
 
         port.onMessage.addListener(message => {
-            this.#friends.forEach(conn => conn.send(message));
+            this.#friends.forEach(friend => friend.connection?.send(message));
         });
 
         this.#peer.on("connection", (conn: Peer.DataConnection) => {
-            this.#handleConnect(conn);
-            // for incoming connections, poll our video and transmit the status
-            setTimeout(() => port.postMessage({ type: MessageType.Poll }), 500);
+            this.onPeerConnectToHost(conn);
         });
     }
 
-    public connect(hostId: string): void {
+    /**
+     * Subscribe to friend changes
+     * @param cb The callback to call when the friends list changes
+     */
+    public subscribeToFriendChanges(cb: (count: number, delta: number) => void): void {
+        this.friendChangeSubscribers.push(cb);
```

Comment:
Per discussion ... we can remove this if we go back to just overwriting an `onChange` function on the class.